modules/draco/src/lib/draco-builder.js

```diff
@@ -68,10 +71,16 @@ export default class DracoBuilder {
   }
 
   _encodePointCloud(pointcloud, options) {
+    const dracoPointCloud = new this.draco.PointCloud();
+
+    if (options.metadata) {
+      this._addGeometryMetadata(dracoPointCloud, options.metadata);
+    }
+
     const attributes = this._getAttributesFromMesh(pointcloud);
 
     // Build a `DracoPointCloud` from the input data
-    const dracoPointCloud = this._createDracoPointCloud(attributes, options);
+    this._createDracoPointCloud(dracoPointCloud, attributes, options);
```

Comment:
Since_createDracoMesh and _createDracoPointCloud no longer actually creates these object, I would rename them to `_buildDracoMesh` / `_buildDracoPointCloud` or something along those lines.

Was it necessary to add the metadata first? I assume that why you made this change?